### PR TITLE
feature(transfer): package creation

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-1167](https://emurgo.atlassian.net/browse/YOMO-1167)

TODO:

- [x] update stories
- [x] package 100% test coverage
- [x] move types to @yoroi/types

[YOMO-1167]: https://emurgo.atlassian.net/browse/YOMO-1167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by michalszorad — APPROVED** [review]

File: apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx [hunk 1]
```diff
@@ -78,6 +84,7 @@ export const ListAmountsToSendScreen = () => {
     track.sendSelectAssetSelected(assetsToSendProperties({tokens, amounts}))
     createUnsignedTx([targets[selectedTargetIndex].entry], {
       onSuccess: (yoroiUnsignedTx) => {
+        console.log('aaa')
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx]:
```suggestion
```

File: packages/transfer/package.json [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    "url": "https://github.com/Emurgo/yoroi.git",
+    "directory": "packages/transfer"
+  },
+  "license": "Apache-2.0",
+  "author": "EMURGO Fintech <support@emurgo.com> (https://github.com/Emurgo/yoroi)",
+  "contributors": [
+    {
+      "name": "Juliano Lazzarotto",
+      "email": "30806844+stackchain@users.noreply.github.com"
+    },
```
Comment (stackchain) [inline packages/transfer/package.json]:
```suggestion
```

File: packages/transfer/src/helpers/mocks.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
+/* istanbul ignore file */
+import {defaultTransferState} from '../translators/reactjs/state/state'
+
+const secondaryTokenId =
+  '698a6ea0ca99f315034072af31eaac6ec11fe8558d3f48e9775aab9d.7444524950'
+
+export const mocks = {
```
Comment (stackchain) [inline packages/transfer/src/helpers/mocks.ts]:
```suggestion
export const mocks = Object.freeze({
```

File: packages/transfer/src/helpers/mocks.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+              ['another.02']: '3',
+              ['more.03']: '4',
+              '': '50000',
+            },
+          },
+        },
+      ],
+    },
+  },
+}
```
Comment (stackchain) [inline packages/transfer/src/helpers/mocks.ts]:
```suggestion
})
```

File: packages/transfer/src/translators/reactjs/provider/TransferProvider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    })
+
+    expect(result.current.targets[0]?.receiver).toEqual<Resolver.Receiver>({
+      resolve: '',
+      as: 'address',
+      selectedNameServer: Resolver.NameServer.Cns,
+      addressRecords: undefined,
+    })
+  })
+  test('addressRecordsFetched', () => {
```
Comment (stackchain) [inline packages/transfer/src/translators/reactjs/provider/TransferProvider.test.tsx]:
```suggestion
  })

  test('addressRecordsFetched', () => {
```

File: yarn.lock [hunk 6]
```diff
@@ -6515,6 +6515,13 @@
   resolved "https://registry.yarnpkg.com/@types/react-native-background-timer/-/react-native-background-timer-2.0.0.tgz#c44c57f8fbca9d9d5521fdd72a8f55232b79381e"
   integrity sha512-y5VW82dL/ESOLg+5QQHyBdsFVA4ZklENxmOyxv8o06T+3HBG2JOSuz/CIPz1vKdB7dmWDGPZNuPosdtnp+xv2A==
 
+"@types/react-native@0.71.6":
```
Comment (stackchain) [inline yarn.lock]: this can be dropped, is no longer required after 0.71

File: packages/types/src/transfer/state.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  Datum as DatumType,
+} from '@emurgo/yoroi-lib'
+
+export type TransferUnsignedTx = TransferTxInfo & {
+  unsignedTx: UnsignedTxType
+}
+
+export type TransferTxInfo = {
+  entries: TransferEntry[]
+  fee: Balance.Amounts
```
Comment (stackchain) [inline packages/types/src/transfer/state.ts]: consume inside types as the original type `BalanceAmounts`

File: packages/types/src/transfer/state.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  change: TransferEntry[]
+  metadata: TransferMetadata
+  staking: TransferStaking
+  voting: TransferVoting
+  governance: boolean
+}
+
+export type TransferEntry = {
+  address: TransferAddress
+  amounts: Balance.Amounts
```
Comment (stackchain) [inline packages/types/src/transfer/state.ts]: :arrow_up:

File: packages/types/src/transfer/state.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  registration?: {
+    votingPublicKey: string
+    stakingPublicKey: string
+    rewardAddress: TransferAddress
+    nonce: number
+  }
+}
+
+export type TransferTarget = {
+  receiver: Resolver.Receiver
```
Comment (stackchain) [inline packages/types/src/transfer/state.ts]: ⬆️

File: packages/types/src/index.ts [hunk 10]
```diff
@@ -279,6 +290,18 @@ export namespace Resolver {
   }
 }
 
+export namespace Transfer {
```
Comment (stackchain) [inline packages/types/src/index.ts:293]: keep only Target, Targets, Entry, as part of Transfer, move everything else to namespace Chain, namespace Cardano -> Chain.Cardano.{Address,UnsignedTx,TxInfo,Metadata,Staking,Voting,Address}

File: packages/transfer/src/translators/reactjs/state/state.ts [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+      case TransferActionType.MemoChanged:
+        draft.memo = action.memo
+        break
+      case TransferActionType.TokenSelectedChanged:
+        draft.selectedTokenId = action.tokenId
+        break
+      case TransferActionType.Reset:
+        draft.selectedTokenId = defaultTransferState.selectedTokenId
+        draft.memo = defaultTransferState.memo
+        draft.unsignedTx = castDraft(defaultTransferState.unsignedTx)
```
Comment (stackchain) [inline packages/transfer/src/translators/reactjs/state/state.ts:35]: castDraft ❤️ ( I've been doing it manually :D )

**Comment by stackchain** [comment]
looking awesome, minor details

**Pull request merged**